lib/ui/painting.dart

```diff
@@ -5521,8 +5563,26 @@ class ImmutableBuffer extends NativeFieldWrapperClass1 {
       instance._init(list, callback);
     }).then((_) => instance);
   }
+
+  /// Create a buffer from the asset with key [assetKey].
+  ///
+  /// Returns `null` if the asset does not exist.
+  static ImmutableBuffer? fromAsset(String assetKey) {
+    final ImmutableBuffer buffer = ImmutableBuffer._(0);
+    bool success = false;
+    buffer._initFromAsset(assetKey, (void result) {
+      success = true;
+    });
+    if (!success) {
+      return null;
```

Comment:
I think this should throw rather than returning null.

For example, if you try to open an asset or file that doesn't exist that should throw. In other cases like this API we do not ignore the error message sent from native.